Both error when install using "npm install phantomjs" and "node ./install.js" on mac

1) npm install phantomjs

Downloading http://cdn.bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.7-macosx.zip
Saving to /var/folders/4q/45xt6phn4nz_q02_nchk_c300000gn/T/phantomjs/phantomjs-1.9.7-macosx.zip

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: connect ETIMEDOUT
    at errnoException (net.js:901:11)
    at Object.afterConnect [as oncomplete](net.js:892:19)
npm ERR! phantomjs@1.9.7-5 install: `node install.js`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the phantomjs@1.9.7-5 install script.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.1.0
npm ERR! command "node" "/usr/local/bin/npm" "install" "phantomjs"
npm ERR! node -v v0.10.24
npm ERR! npm -v 1.4.9
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR! not ok code 0

2) git clone the source code and node ./install.js

module.js:340
    throw err;
          ^
Error: Cannot find module 'adm-zip'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/...../phantomjs/install.js:9:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)

--- a/README.md
+++ b/README.md
@@ -16,6 +16,13 @@ Or grab the source and
 node ./install.js
 ```
 
+To use a mirror of the phantomjs binaries, set `$PHANTOMJS_CDNURL`,
+default is `http://cdn.bitbucket.org/ariya/phantomjs/downloads`
+
+```shell
+PHANTOMJS_CDNURL=http://cnpmjs.org/downloads npm install phantomjs
+```
+
 What this is really doing is just grabbing a particular "blessed" (by
 this module) version of Phantom. As new versions of Phantom are released
 and vetted, this module will be updated accordingly.

--- a/install.js
+++ b/install.js
@@ -21,7 +21,8 @@ var url = require('url')
 var util = require('util')
 var which = require('which')
 
-var downloadUrl = 'http://cdn.bitbucket.org/ariya/phantomjs/downloads/phantomjs-' + helper.version + '-'
+var cdnUrl = process.env.PHANTOMJS_CDNURL || 'http://cdn.bitbucket.org/ariya/phantomjs/downloads'
+var downloadUrl = cdnUrl + '/phantomjs-' + helper.version + '-'
 
 var originalPath = process.env.PATH
 

